pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -774,6 +781,21 @@ func initResources() (corev1.ResourceRequirements, error) {
 	return resources, nil
 }
 
+func initSecurityContext() (*corev1.SecurityContext, error) {
```

Comment:
Instead of calling this method each time for each init container, which will require unmarshalling json string each time, I think it will be better to define the init container security context as a property on [the webhook struct](https://github.com/DataDog/datadog-agent/blob/5b9a9a8389e3afbc8362a8e56852f0a0a7349070/pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go#L138-L148).

After that, you might initialise this property in the [`NewWebhook` function](https://github.com/DataDog/datadog-agent/blob/5b9a9a8389e3afbc8362a8e56852f0a0a7349070/pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go#L151-L170). 
It can look like this:

```
// NewWebhook returns a new Webhook
func NewWebhook(wmeta workloadmeta.Component) (*Webhook, error) {
	filter, err := apmSSINamespaceFilter()
	if err != nil {
		return nil, err
	}

	containerRegistry := mutatecommon.ContainerRegistry("admission_controller.auto_instrumentation.container_registry")

   initSecurityContext, err := parseInitSecurityContext()
   if err != nil {
       return nil, err
   }

	return &Webhook{
		name:              webhookName,
		isEnabled:         config.Datadog().GetBool("admission_controller.auto_instrumentation.enabled"),
		endpoint:          config.Datadog().GetString("admission_controller.auto_instrumentation.endpoint"),
		resources:         []string{"pods"},
		operations:        []admiv1.OperationType{admiv1.Create},
		filter:            filter,
        initSecurityContext: initSecurityContext,
		containerRegistry: containerRegistry,
		pinnedLibraries:   getPinnedLibraries(containerRegistry),
		wmeta:             wmeta,
	}, nil
}
```

I this, I renamed your function to `parseInitSecurityContext`.

Finally, you can directly access security context as a property on the webhook object.

Reply:
Hi @adel121 - thanks for the review! 🙏 I think your suggestion makes a lot of sense, since unmarshalling can be costly - thanks for the pointers ^^
I've implemented it, and adjusted the tests 👍